src/dos/drive_iso.cpp

```diff
@@ -64,14 +64,18 @@ isoFile::isoFile(isoDrive *drive, const char *name, FileStat_Block *stat, Bit32u
 bool isoFile::Read(Bit8u *data, Bit16u *size) {
 	if (filePos + *size > fileEnd)
 		*size = (Bit16u)(fileEnd - filePos);
-	
-	Bit16u nowSize = 0;
-	int sector = filePos / ISO_FRAMESIZE;
-	Bit16u sectorPos = (Bit16u)(filePos % ISO_FRAMESIZE);
-	
-	if (sector != cachedSector) {
-		if (drive->readSector(buffer, sector)) cachedSector = sector;
-		else { *size = 0; cachedSector = -1; }
+
+	uint16_t nowSize = 0;
+	uint32_t sector = filePos / ISO_FRAMESIZE;
+	uint16_t sectorPos = (Bit16u)(filePos % ISO_FRAMESIZE);
```

Comment:
`static_cast<uint16_t>` to avoid C-cast and `Bit16u`.

I would think about adding a static assert in this code as well (to assure, that modulo operation can be safely casted to `uint16_t` without overflow):
```C++
static_assert(ISO_FRAMESIZE <= std::numeric_limits<uint16_t>::max(), "");
```

BTW, I know we use `std::numeric_limits` for some time now, but how about we switch to using C99 constants defined in `cinttypes` when we need max value for an integer type? In this case it would look like this:
```C++
static_assert(ISO_FRAMESIZE <= UINT16_MAX, "");
```

Reply:
Great catch; I think there might be a couple more static assert opportunities in the other cleanup PRs too - I will take another pass through them too.

Yes - those limits are a lot easier to remember don't blast the line length out :-) I had actually started using them in the other two cleanup PRs, coincidentally.  

Does it make sense to do a PR that flips the project over en-mass? (A couple recursive seds would do the trick)